### PR TITLE
feat: add EC2 launch template for EKS worker nodes

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Add an EC2 launch template that enables IMDSv2 for the worker nodes.  This is a [security best practice](https://aws.amazon.com/about-aws/whats-new/2020/08/amazon-eks-supports-ec2-instance-metadata-service-v2/) but unfortunately not part of the default launch template.

The values were determined by matching the current default launch template used by the worker nodes:
```sh
# Get launch template ID
aws ec2 describe-instances \
    --instance-ids $INSTANCE_ID \
    --query 'Reservations[0].Instances[0].Tags[?Key==`aws:ec2launchtemplate:id`]'

# Get launch template
aws ec2 describe-launch-template-versions \
    --launch-template-id $LAUNCH_TEMPLATE_ID \
    --versions '$Default'
```

# Test instructions | Instructions pour tester la modification

1. Merge the PR and confirm that the launch template is created properly.
2. Trigger the creation of the secondary node group and confirm that worker nodes are created as expected.
3. Switch the EKS cluster to use the secondary node group and run smoke and integration tests.

# Related
- https://github.com/cds-snc/platform-core-services/issues/489

### CODE REVIEW

File: aws/eks/eks.tf [hunk 1]
```diff
@@ -127,6 +130,33 @@
...
+    http_endpoint               = "enabled"
+    http_put_response_hop_limit = 2
+    http_tokens                 = "required"
+  }
+
+  tags = {
+    Name       = "notification-canada-ca"
+    CostCenter = "notification-canada-ca-${var.env}"
+  }
+}
```
[the remark this answers is not in the document]
Comment (patheard) [reply aws/eks/eks.tf:158]: Hop limit > 1 is required for containerized applications in EKS as they have a segmented network which counts as a hop.

**Comment by patheard** [comment]
This has now been tested in the dev account by creating the secondary node group.  EC2 instances were created and joined the worker node pool without issue.

Comparing the primary and secondary node group instances also showed only one difference, which was IMDSv2 being enabled on the secondary node group instances.

Next week we'll test how the dev cluster behaves with only IMDSv2 worker nodes.

**Review by ben851 — APPROVED** [review]
Half a year later, LGTM

**Comment by patheard** [comment]
image

**Pull request merged**